OpenSR/scripts/gui/tabs/GalaxyTab.as

```diff
@@ -274,8 +275,10 @@ class GalaxyTab : Tab {
 
 			bool isFloating = false;
 			file >> isFloating;
+			bool valid = true;
+			file >> valid;
 
-			if(obj !is null) {
+			if(obj !is null && obj.valid && valid) {
```

Comment:
Getting rid of ` obj.valid` is not possible? Like this only works in this specific combination of conditions?

Perhaps put a small comment here why just checking `obj.valid` alone is not enough on load (seeing as in essence `valid` is just `obj.valid` being stored in the save).

Reply:
My guess is object validity might not be saving or recalculating correctly for a ship that's already been destroyed. Ie before the save we have a handle to a now dead ship, so possibly that ship doesn't have a ShipScript anymore which would mean it doesn't get saved to file properly. Then on reload it's an object but no longer a ship?

Reply:
More likely, the object's validity isn't recomputed in time for us to check it here. Maybe when *everything* has been deserialized, or on the first tick...

However, that wasn't the question. The question was, why do you have `obj.valid && valid` when, by your own admission, `obj.valid` doesn't contain the data you need?

On another note: You could have checked object validity *before* (de)serializing the whole popup.

Reply:
Oh. That's just future proofing. If an object can be invalid before saving and appear valid after, what's to say the reverse can't be true?

And yes the order of serialising and deserialising the check could be moved around I suppose, but to me it made sense to extend the existing null check vanilla has so I've done it in this order to facilitate that.

Reply:
Then definitely leave a comment on it.

Anyway, it makes more sense to me to minimize the amount of work (and disk space) used by the whole process.

Reply:
I would be very surprised if star ruler saved the same object to disk multiple times. I suspect it is smart enough to save an identifier for all the subsequent saves.

Reply:
I'm talking about saving the popup! 😛

Reply:
feel free to add another commit to save a few bytes by not saving a dead popup then :P

Reply:
I will! And I'll fix the revision number, too!

... Sometime soonish.

Reply:
Meh. I guess I never did get around to it.